SplunkRumWorkspace/SplunkRum/SplunkRum/SpanDb.swift

```diff
@@ -0,0 +1,281 @@
+//
+/*
+Copyright 2022 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import SQLite3
+
+fileprivate func sqliteError(code: Int32) -> String {
+    return String(cString: sqlite3_errstr(code))
+}
+
+fileprivate let SQLITE_TRANSIENT = unsafeBitCast(-1, to: sqlite3_destructor_type.self)
+
+class SpanDb {
+    let databasePath: String
+    private var lock: NSLock = NSLock()
+    internal var db_: OpaquePointer?
+    private var insertStmt_: OpaquePointer?
+    private var fetchStmt_: OpaquePointer?
+    private var sizeStmt_: OpaquePointer?
+    private var initialized: Bool = false
+
+    init(path: String? = nil) {
+        self.databasePath = path ?? SpanDb.makeDatabasePath() ?? ":memory:"
+
+        var status = sqlite3_open(databasePath, &db_)
+
+        if status != SQLITE_OK {
+            print("Failure opening \(databasePath): \(sqliteError(code: status))")
```

Comment:
I've tried to use the term `SplunkRum` in any log messages like this before (e.g., `print("SplunkRum: Please call SplunkRum.initialize only on the main thread")`.  Can you please audit these and make sure that anybody staring at some random logs can know this stuff came from us?

Reply:
👍 Done: https://github.com/signalfx/splunk-otel-ios/pull/82/commits/22f4b97fbf3cda0dfe9990795ef24f4c1e9a140f